cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,54 @@
 /// <reference types='cypress' />
 
+const { add } = require('lodash');
+const { generateUser } = require('../support/generate');
+
 describe('Student Registration page', () => {
   before(() => {
-    
+    cy.visit('https://demoqa.com/automation-practice-form');
   });
 
-  it('', () => {
-    
+  it('Fill all fields in forms except "picture"', () => {
+    const { firstName, lastName, email, phoneNumber, dateOfBirth, address, subject, gender } = generateUser();
+    cy.get('#firstName')
+      .type(firstName);
+    cy.get('#lastName')
+      .type(lastName);
+    cy.get('#userEmail')
+      .type(email);
+    cy.get(`#gender-radio-${gender}`)
+      .check({ force: true });
+    cy.get('#userNumber')
+      .type(phoneNumber);
+    cy.get('#dateOfBirthInput')
+      .type(dateOfBirth);
+    cy.get('.css-1hwfws3')
+      .type(subject)
+    cy.get(`#hobbies-checkbox-${Math.ceil(Math.random() * 3)}`)
+      .check({ force: true });
+    cy.get('#currentAddress')
+      .type(address);
+    cy.get('#state').click()
+      .get(`#react-select-3-option-${Math.floor(Math.random() * 4)}`)
```

Comment:
Using 'Math.random()' inside test might make the test non-deterministic. Consider assigning a fixed value instead.